### PR TITLE
DOC: Fix typos and grammar

### PR DESCRIPTION
**Pull request by jhlegarreta**
- DOC: Fix typos and grammar in docstrings
- TEST: Fix typos in test method names
- STYLE: Fix typos in local variable naming

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Test errors are unrelated (related to the `dipy.io.bvectxt` being deprecated).

File: dipy/direction/tests/test_bootstrap_direction_getter.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 
 def test_bdg_initial_direction():
-    """This test the number of inital direction."
+    """This tests the number of initial directions."
```
Comment (skoudoro) [inline dipy/direction/tests/test_bootstrap_direction_getter.py:24]: not sure about this one

**Review by skoudoro — COMMENTED** [review]
thank you for this @jhlegarreta.

That's great! See below, I just have 1 doubt, otherwise this is good to go

File: dipy/direction/tests/test_bootstrap_direction_getter.py (hunk 1 above)
Comment (skoudoro) [reply dipy/direction/tests/test_bootstrap_direction_getter.py:24]: meaning `test` singular

Comment (arokem) [reply dipy/direction/tests/test_bootstrap_direction_getter.py:24]: I can confirm that the proposed change is 👍

Comment (jhlegarreta) [reply dipy/direction/tests/test_bootstrap_direction_getter.py:24]: :+1: Strictly speaking, I'd say that test methods do not require docstrings. Additionally, if I'm not mistaken PEP8 discourages starting docstrings with wording like "This method/class" etc. At the same time, if the naming or the test contents are not clear enough (which are indicative of bad naming wrt the acronyms/context of the package, or too many things done in a test), I'd opt for a simple comment like `# Test the number of initial directions`. But that is beyond the scope of this PR.

Thanks both for the reviews.

**Comment by skoudoro** [comment]
Failing CI's are not related, so I am going ahead and merging this.

Thanks @jhlegarreta

**Pull request merged**